### DISCUSSION TITLE
[Bug]: Storybook 8 union type arguments values are shown as "union" instead of individual values.

### DISCUSSION THREAD

**Discussion by hailwood** [storybookjs/storybook#26427] [Help]
### Describe the bug

In the docs it shows that we should be able to see the indivisual values for a union type - 
![image](https://github.com/storybookjs/storybook/assets/709773/8f7dc348-8edc-49bf-8156-25a53929afeb)

However in a brand new project we're seeing "union" instead of the type - 
![image](https://github.com/storybookjs/storybook/assets/709773/29fb4c90-7199-4bf3-b469-b9f9e45d48f5)

It's still infering the options correctly, but the value showed is not collapsed.

### To Reproduce

Brand new stackblitz, no changes have been made.

https://stackblitz.com/edit/github-hyzo8z?file=src%2Fstories%2FButton.tsx&preset=node

### System

```bash
Storybook Environment Info:

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm <----- active
    pnpm: 8.15.3 - /usr/local/bin/pnpm
  npmPackages:
    @storybook/addon-essentials: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/addon-interactions: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/addon-links: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/addon-onboarding: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/blocks: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/react: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/react-vite: ^8.0.0-rc.3 => 8.0.0-rc.3 
    @storybook/test: ^8.0.0-rc.3 => 8.0.0-rc.3 
    storybook: ^8.0.0-rc.3 => 8.0.0-rc.3
```

**Comment by vanessayuenn** [comment]
We changed how react components are parsed for performance reasons -- the default parser is now [`react-docgen`](https://github.com/reactjs/react-docgen) instead of [`react-docgen-typescript`](https://github.com/styleguidist/react-docgen-typescript). You can revert to old behaviour by [configuring the `reactDocgen` option](https://storybook.js.org/docs/api/main-config-typescript#reactdocgen).

**Reply by bromanazzi** [reply]
I'm confused, it can correctly parse and display all union options in the control, but in the description it only shows `union`? It can't display the same values as the control?
